platform/eureka/client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eureka
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type application struct {
+	Name      string      `json:"name"`
+	Instances []*instance `json:"instance"`
+}
+
+type instance struct {
+	Hostname   string   `json:"hostName"`
+	App        string   `json:"app"`
+	IPAddress  string   `json:"ipAddr"`
+	Status     string   `json:"status"`
+	Port       *port    `json:"port,omitempty"`
+	SecurePort *port    `json:"securePort,omitempty"`
+	Metadata   metadata `json:"metadata,omitempty"`
```

Comment:
Eureka's VIPs aren't the same as k8s service addresses. VIP address can be used to obtain hostname/port pairs using the Eureka client, but the requests themselves are made to the hostname, not the VIP address.

See: https://github.com/Netflix/eureka/wiki/Understanding-eureka-client-server-communication#eureka-client-operations

We may have it in the future, but until then our linters get mad because it is an unused variable. :)